Services/Services/InterviewService.cs

```diff
@@ -0,0 +1,128 @@
+                    interview.Type);
+
+            return interviewDTO;
+        }
+
+        public IEnumerable<InterviewDTO> GetInterviews(Paging paging, SortOrder sortOrder, InterviewFiltringDTO interviewFiltringDTO)
+        {
+            IQueryable<Interview> interviews = _interviewRepository.GetAll();
+
+            interviews = interviews.Where(i => (i.Date >= interviewFiltringDTO.FromDate && i.Date <= interviewFiltringDTO.ToDate));
+
+            if (interviewFiltringDTO.CandidateId.HasValue)
+            {
+                interviews = interviews.Where(s => s.CandidateId == interviewFiltringDTO.CandidateId);
+            }
+            if (interviewFiltringDTO.WorkerId.HasValue)
+            {
+                interviews = interviews.Where(s => s.WorkerId == interviewFiltringDTO.WorkerId);
+            }
+            if (!String.IsNullOrEmpty(interviewFiltringDTO.Type))
+            {
+                interviews = interviews.Where(s => s.Type.Equals(interviewFiltringDTO.Type));
+            }
+
+            foreach (KeyValuePair<string, string> sort in sortOrder.Sort)
+            {
+                if (sort.Key.ToLower() == "date")
+                {
+                    if (sort.Value.ToUpper() == "DESC")
+                    {
+                        interviews = interviews.OrderByDescending(i => i.Date);
+                    }
+                    else
+                    {
+                        interviews = interviews.OrderBy(i => i.Date);
+                    }
+                }
+                if (sort.Key.ToLower() == "candidateid")
+                {
+                    if (sort.Value.ToUpper() == "DESC")
+                    {
+                        interviews = interviews.OrderByDescending(i => i.CandidateId);
+                    }
+                    else
+                    {
+                        interviews = interviews.OrderBy(i => i.CandidateId);
+                    }
+                }
+                if (sort.Key.ToLower() == "userid")
+                {
+                    if (sort.Value.ToUpper() == "DESC")
+                    {
+                        interviews = interviews.OrderByDescending(i => i.WorkerId);
+                    }
+                    else
+                    {
+                        interviews = interviews.OrderBy(i => i.WorkerId);
+                    }
+                }
+                if (sort.Key.ToLower() == "type")
+                {
+                    if (sort.Value.ToUpper() == "DESC")
+                    {
+                        interviews = interviews.OrderByDescending(i => i.Type);
+                    }
+                    else
+                    {
+                        interviews = interviews.OrderBy(i => i.Type);
+                    }
+                }
+            }
+
+            IEnumerable<InterviewDTO> resutl2 = (from i in interviews
+                                                  from c in _candidateRepository.GetAll()
```

Comment:
You can use lambda expressions to get this data. You can use `Include` to have Candidate and Worker that is connected to Interview